spec/streamflow/pm.md

```diff
@@ -0,0 +1,408 @@
+
+If the broadcaster's reserve was previously frozen and the freeze period is over, `fundReserve` will move any remaining funds in the old reserve into a new reserve in addition to any funds sent as a part of the `fundReserve` call.
+
+```
+/**
+ * @dev Adds ETH to the caller's reserve
+ */
+function fundReserve() external payable;
+```
+
+`fundReserve` will revert under the following conditions:
+
+- The caller's reserve is currently frozen
+
+#### fundDepositAndReserve
+
+Broadcasters call `fundDepositAndReserve` in order to add ETH to both their deposit and reserve in a single atomic transaction.
+
+```
+/**
+ * @dev Adds ETH to the caller's deposit and reserve
+ * @param _depositFunds ETH to add to the caller's deposit
+ * @param _reserveFunds ETH to add to the caller's reserve
+ */
+function fundDepositAndReserve(uint256 _depositFunds, uint256 _reserveFunds) external payable;
+```
+
+`fundDepositAndReserve` will revert under the following conditions:
+
+- `_depositFunds + _reserveFunds` is not equal to the amount of ETH sent with the `fundDepositAndReserve` call
+- The deposit funding (which should use the same internal logic as `fundDeposit`) process halts with a revert
+- The reserve funding (which should use the same internal logic as `fundReserve`) process halts with a revert
+
+
+#### unlock
+
+Broadcasters call `unlock` to start the unlock period. They are able to withdraw their funds after the unlock period is over. Broadcasters are unable to call `unlock` if their reserve is currently frozen.
+
+```
+/**
+ * @dev Initiates the unlock period for the caller. This function can only be called
+ * if the caller is not already in the unlock period
+ */
+function unlock() public;
+```
+
+`unlock` will revert under the following conditions:
+
+- The caller's deposit and reserve are both empty 
+- The caller already initiated the unlock period
+- The caller's funds are already unlocked
+- The caller's reserve is currently frozen
+
+#### cancelUnlock
+
+Broadcasters call `cancelUnlock` to cancel the unlock period.
+
+```
+/**
+ * @dev Cancels the unlocking period for the caller. This function can only be called
+ * if the caller is in the unlock period
+ */
+function cancelUnlock() public;
+```
+
+`cancelUnlock` will revert under the following conditions:
+
+- The caller is not in the unlock period
+- The caller's reserve is currently frozen
+
+#### withdraw
+
+Broadcasters call `withdraw` to withdraw funds after waiting through either the unlock or freeze period.
+
+```
+/**
+ * @dev Withdraws all ETH from the caller's deposit and reserve. This function can only be
+ * called if the caller's funds are unlocked 
+ */
+function withdraw() public;
+```
+
+`withdraw` will revert under the following conditions:
+
+- The caller's deposit and reserve are both empty
+- The caller's funds are not unlocked
+
+## Actions
+
+### Broadcaster State Transition Diagram
+
+![broadcaster-protocol-states](images/broadcaster-protocol-states.png)
```

Comment:
Does B start at "unlocked"?  If so, does B have to unlock before "refilling" the deposit?

Reply:
> Does B start at "unlocked"?

Yes. I can add a note to clarify this.

> If so, does B have to unlock before "refilling" the deposit?

Yes. My current thinking is that restricting B from adding additional funds (for both its deposit and reserve) when B is in the `Frozen` state simplifies state transition logic. 

In theory, B could add additional funds to a *new* reserve while its *old* reserve is in the freeze period (so that eligible Os can claim from it). However, allowing for two reserves to exist simultaneously feels like it adds additional complexity without much additional benefit.

B could also add additional funds to its deposit while its *old* reserve is in the freeze period, but if B already has to wait until its *old* reserve is out of the freeze period in order to create a *new* reserve (which would allow B to work with Os again), then I don't see much additional benefit in allowing B to add additional funds to its deposit while it in the `Frozen` state (especially since B can add additional funds to both its deposit and its *new* reserve in a single atomic transaction after the freeze period).

Reply:
Added a note about broadcasters starting in the `Unlocked` state [here](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R256). Let me know if this is clear enough.

Reply:
Hmm... Maybe I'm missing something... I understand we don't want B to mess with funds in the `Frozen` state, but why can't B call `fundDeposit()` or `fundReserve()` in the `Locked` state?

Reply:
B can call `fundDeposit()`/`fundReserve()`/`fundDepositAndReserve()` while in the `Locked` state. I see how the diagram might've implied otherwise...I just pushed an update to the diagram to indicate that if B calls `fundDeposit()`/`fundReserve()`/`fundDepositAndReserve` while in the `Locked` state it will remain in the `Locked` state.